File: api/julia/api_to_json.jl
using HTTP, FileIO

function main()
    url = "https://randomuser.me/api/"
    file_path = "data.json"

    try
        response = HTTP.get(url)
        if response.status == 200
            body = String(response.body)
            FileIO.write(file_path, body)
            println("File saved successfully.")
        else
            println("Error requesting URL: $(response.status)")
        end
    catch e
        println("An error occurred: $(e)")
    end
end

main()

File: api/rust/api_to_json.rs
use std::fs::File;
use std::io::Write;
use std::io::Read;
use reqwest;

fn main() {
    let url = "https://randomuser.me/api/";
    let file_path = "data.json";

    match reqwest::blocking::get(url) {
        Ok(mut response) => {
            let mut body = Vec::new();
            response.read_to_end(&mut body).unwrap();

            match File::create(file_path) {
                Ok(mut file) => {
                    file.write_all(&body).unwrap();
                    println!("File saved successfully.");
                }
                Err(e) => {
                    eprintln!("Error creating file: {}", e);
                }
            }
        }
        Err(e) => {
            eprintln!("Error requesting URL: {}", e);
        }
    }
}
